### PR TITLE
Plant mutants harvest vitamins and nutrition from soil and sunlight

### LINKED ISSUE #29241: Plant mutants suffering vitamin deficiency
**Describe the bug**  
The description for plant mutations suggests that they could survive off soil and sunlight alone. Doing exactly this on my character for over a month has left me with some massive vitamin deficiencies. A peek at the debug menu tells me I'm around -4k for most of my vitamins.

**To Reproduce**  
1. Be a plant mutant with any of the root mutations.
2. Only use that to satisfy your food and drink needs for a long period of time.
3. Suffer from massive vitamin deficiency.

**Expected behavior**  
I thought rooting in dirt would cover my food crisis, but alas it did not. The amount of food I have to eat to address my deficiencies seems to completely offset the benefit of rooting.

**Versions and configuration(please complete the following information):**
 - OS: Windows 8.1
 - Game Version: 0.D-1319-gdca9d26
 - Graphics version: Tiles
 - Mods loaded: Dark Days Ahead, Garden Pots, Makeshift Items Mod, More Survival Tools, Mutant NPCs, More Locations, Urban Development, Fuji's More Buildings, Boats, Folding Parts Pack, Vehicle Additions Pack, Tanks and Other Vehicles, Filthy Clothing, Manual Bionic Installation, Safe Autodoc

### PR DESCRIPTION
**Pull request by kevingranade**
Refactored from #30042 to make gains immediate instead of indirecting through new vitamin counter system.

#### Summary
SUMMARY: Bugfixes "Plant mutants now absorb vitamins as well."

#### Purpose of change
Fixes #29241
Obsoletes #29907 and #30042 
Plant mutants now have a more pervasive source of sustenance.

#### Describe the solution
Every turn that a plant mutant spends in eligible sunlight or rooted, they have a chance to generate vitamins and other nutrition.

#### Describe alternatives you've considered
#30042 does the same thing but by adding some unnecessary pseudo-vitamin infrastructure.

#### Additional context
Plant mutations can provide some sustenance, but due to extreme metabolic differences between plants and animals, plant mutations are unable to provide all the nutrition needed by an active creature.

### CODE REVIEW

File: src/player.cpp [hunk 1]
```diff
@@ -5785,7 +5784,30 @@ void player::suffer()
         }
     }
 
-    if( has_trait( trait_LEAVES ) && g->is_in_sunlight( pos() ) && one_in( 600 ) ) {
+    const auto player_local_temp = g->weather.get_temperature( pos() );
```
Comment (ifreund) [inline src/player.cpp]:
```suggestion
    const int player_local_temp = g->weather.get_temperature( pos() );
```

**Comment by KorGgenT** [comment]
plants will remain hungry or peckish, with empty stomachs.

**Comment by kevingranade** [comment]
Considering the fact that they still have stomachs, and that they're going to be starving to death if they don't eat, that sounds about right.

File: src/player.cpp [hunk 2]
```diff
@@ -7493,24 +7515,21 @@ void player::rooted_message() const
...
-                mod_stored_nutr( -1 );
-            }
-            if( get_thirst() > -20 ) {
-                mod_thirst( -1 );
-            }
-            mod_healthy_mod( 5, 50 );
+        if( one_in( 96 ) ) {
+            vitamin_mod( vitamin_id( "iron" ), 1, true );
+            vitamin_mod( vitamin_id( "calcium" ), 1, true );
+	}
```
Comment (ZhilkinSerg) [inline src/player.cpp]:
```suggestion
        }
```

File: src/player.cpp [hunk 3]
```diff
@@ -4443,6 +4446,11 @@ needs_rates player::calc_needs_rates()
         }
     }
 
+    if( has_trait( trait_TRANSPIRATION ) ) {
+        // Tranpiration, the act of moving nutrients with evaporating water, can take a very heavy toll on your thirst when it's really hot.
```
Comment (Night-Pryanik) [inline src/player.cpp]:
```suggestion
        // Transpiration, the act of moving nutrients with evaporating water, can take a very heavy toll on your thirst when it's really hot.
```

File: src/player.cpp [hunk 4]
```diff
@@ -5785,7 +5784,30 @@ void player::suffer()
         }
     }
 
-    if( has_trait( trait_LEAVES ) && g->is_in_sunlight( pos() ) && one_in( 600 ) ) {
+    const int player_local_temp = g->weather.get_temperature( pos() );
+    double sleeve_factor = armwear_factor();
+    bool has_hat = wearing_something_on( bp_head );
```
Comment (Night-Pryanik) [inline src/player.cpp]:
```suggestion
    const bool has_hat = wearing_something_on( bp_head );
```

File: src/player.cpp [hunk 5]
```diff
@@ -5785,7 +5784,30 @@ void player::suffer()
...
+    bool leafy = has_trait( trait_LEAVES ) || has_trait( trait_LEAVES2 ) || has_trait( trait_LEAVES3 );
```
Comment (Night-Pryanik) [inline src/player.cpp]:
```suggestion
    const bool leafy = has_trait( trait_LEAVES ) || has_trait( trait_LEAVES2 ) || has_trait( trait_LEAVES3 );
```

File: src/player.cpp [hunk 6]
```diff
@@ -5785,7 +5784,30 @@ void player::suffer()
...
+    bool leafier = has_trait( trait_LEAVES2 ) || has_trait( trait_LEAVES3 );
```
Comment (Night-Pryanik) [inline src/player.cpp]:
```suggestion
    const bool leafier = has_trait( trait_LEAVES2 ) || has_trait( trait_LEAVES3 );
```

File: src/player.cpp [hunk 7]
```diff
@@ -5785,7 +5784,30 @@ void player::suffer()
...
+    bool leafiest = has_trait( trait_LEAVES3 );
```
Comment (Night-Pryanik) [inline src/player.cpp]:
```suggestion
    const bool leafiest = has_trait( trait_LEAVES3 );
```

**Pull request merged**